config/routes.rb

```diff
@@ -42,6 +42,11 @@ def matches?(request)
       get     "/resources/:account"                   => "resources#index"
       get     "/resources"                            => "resources#index"
 
+      # NOTE: the order of these routes matters: we need the expire 
+      #       route to come first.
+      constraints account: /[^\/]*/, kind: /[^\/]+/, identifier: /.+/ do
+        post    "/secrets/:account/:kind/:identifier/expirations" => "secrets#expire"
```

Comment:
A couple of things on this:

- First, if just calling the endpoint simply expires the given secret. Then I think this should just be `/expire`.

- Second, if this were to accept a parameter to specify the new expiration (which is what I think @apotterri was asking about), then I think this should be change to the singular `/expiration`

- Third and finally, while I like how you've made the path work here. I think we need to either change the other routes to match (namely the role membership and member routes), or this one should match the others and use a query parameter constraint instead.

Reply:
I though about having the endpoint be `expire`, components of the path are supposed to be nouns, not verbs. The method you're using is supposed to be the verb.

Yes, I was suggesting accepting a parameter to specify the new expiration. if we did that, changing to `expiration` makes sense to me.

Reply:
To summarize a slack discussion. I'm dropping my request for changing the semantics of the endpoint, but only the syntax (`?expirations` instead of `/expirations`) so that the endpoint is consistent with how sub-resources are handled with Roles.